lib/msal-common/src/cache/entities/AccountEntity.ts

```diff
@@ -72,6 +72,32 @@ export class AccountEntity {
         }
     }
 
+    /**
+     * Returns the AccountInfo interface for this account.
+     */
+    getAccountInfo(): AccountInfo {
+        return {
+            homeAccountId: this.homeAccountId,
+            environment: this.environment,
+            tenantId: this.realm,
+            username: this.username
+        };
+    }
+
+    /**
+     * Generates account key from interface
+     * @param accountInterface
+     */
+    static generateAccountCacheKey(accountInterface: AccountInfo): string {
+        const accountKey = [
+            accountInterface.homeAccountId,
+            accountInterface.environment || "",
+            accountInterface.tenantId || "",
```

Comment:
Do we need to include username here as well?

Reply:
username isn't part of the key, it is a part of the value.

Reply:
That's confusing to me because it's passed to this function on line 51 and also returned in `getAccountInfo`

Reply:
Resolved offline. Agreed to add comments around Key,Value for Account